src/extension_instance.ml

```diff
@@ -150,21 +189,35 @@ end = struct
       LanguageClient.Executable.create ~command ~args ~options ()
   ;;
 
-  let check_ocaml_lsp_available sandbox =
-    let ocaml_lsp_version sandbox =
-      Sandbox.get_command sandbox "ocamllsp" [ "--version" ]
-    in
-    let cwd =
-      match Workspace.workspaceFolders () with
-      | [ cwd ] -> Some (cwd |> WorkspaceFolder.uri |> Uri.fsPath |> Path.of_string)
-      | _ -> None
+  let suggest_to_install_ocaml_lsp_server () =
+    let open Promise.Syntax in
+    let install_lsp_text = "Install OCaml-LSP server" in
+    let select_different_sandbox = "Select a different Sandbox" in
+    let+ selection =
+      Window.showInformationMessage
+        ~message:
+          "Failed to start the language server. `ocaml-lsp-server` is not installed in \
+           the current sandbox."
+        ~choices:
+          [ install_lsp_text, `Install_lsp; select_different_sandbox, `Select_sandbox ]
+        ()
     in
-    Cmd.output ?cwd (ocaml_lsp_version sandbox)
-    |> Promise.Result.fold
-         ~ok:(fun (_ : string) -> ())
-         ~error:(fun (_ : string) ->
-           "Sandbox initialization failed: `ocaml-lsp-server` is not installed in the \
-            current sandbox.")
+    match selection with
+    | Some `Install_lsp ->
+      let (_ : Ojs.t option Promise.t) =
+        Vscode.Commands.executeCommand
+          ~command:Extension_consts.Commands.install_ocaml_lsp_server
+          ~args:[]
+      in
+      ()
+    | Some `Select_sandbox ->
+      let (_ : Ojs.t option Promise.t) =
+        Vscode.Commands.executeCommand
+          ~command:Extension_consts.Commands.select_sandbox
+          ~args:[]
+      in
+      ()
```

Comment:
Same here, what happens if we do wait for these promises to fulfill instead of ignoring them ?